Algorithm.CSharp/VolumeRenkoConsolidatorAlgorithm.cs

```diff
@@ -0,0 +1,168 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+using QuantConnect.Indicators;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Demostrates the use of <see cref="VolumeRenkoConsolidator"/> for creating constant volume bar
+    /// </summary>
+    /// <meta name="tag" content="renko" />
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="consolidating data" />
+    public class VolumeRenkoConsolidatorAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _spy, _ibm;
+        private VolumeRenkoConsolidator _tradebarVolumeConsolidator, _tickVolumeConsolidator;
+        private SimpleMovingAverage _sma = new SimpleMovingAverage(10);
+
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 7);
+            SetEndDate(2013, 10, 11);
+            SetCash(100000);
+
+            _spy = AddEquity("SPY", Resolution.Minute).Symbol;
+            _tradebarVolumeConsolidator = new VolumeRenkoConsolidator(1000000);
+            _tradebarVolumeConsolidator.DataConsolidated += (sender, bar) => {
+                _sma.Update(bar.EndTime, bar.Value);
+                Debug($"SPY {bar.Time} to {bar.EndTime} :: O:{bar.Open} H:{bar.High} L:{bar.Low} C:{bar.Close} V:{bar.Volume}");
+                if (bar.Volume != 1000000)
+                {
+                    throw new Exception("Volume of consolidated bar does not match set value!");
+                }
+            };
+
+            _ibm = AddEquity("IBM", Resolution.Tick).Symbol;
+            _tickVolumeConsolidator = new VolumeRenkoConsolidator(1000000);
+            _tickVolumeConsolidator.DataConsolidated += (sender, bar) => {
```

Comment:
Think would be nice to add an assertion that `_tickVolumeConsolidator.DataConsolidated` was called at least once